pipeline/metadata/hyperquack_outcome.py

```diff
@@ -0,0 +1,170 @@
+"""Classify outcomes for roundtrip measurements."""
+
+from __future__ import absolute_import
+
+from typing import List, Optional
+
+
+def _status_mismatch(received_status: Optional[str]) -> str:
+  """
+  Args:
+    received_status: status string like "200 OK" or None
+
+  Returns:
+    outcome like "content/status_mismatch" or "content/status_mismatch:404"
+  """
+  if received_status is None or received_status == "":
+    return "content/status_mismatch"
+  return "content/status_mismatch:" + received_status[:3]
+
+
+def _get_first_error(errors: Optional[str]) -> Optional[str]:
+  """
+  Args:
+    errors: string like "error a; error b; error c" or just "error a"
+
+  Feturns
+    "error a"
+  """
+  if errors is None:
+    return None
+
+  if ';' in errors:
+    return errors[:errors.index(';')]
+  return errors  # single error
+
+
+def _has_akamai_server_header(received_headers: List[str]) -> bool:
```

Comment:
I wish the headers were properly structured. I'd highly recommend that change for the data dumps.
Something like `STRUCT(name: "Server", value: "AkamaiGHost")`

Then you can write code like `if (headers.get("Server") in [...])`